### PR TITLE
Support the new `panic!()` macro

### PR DESCRIPTION
**Pull request by jonas-schievink**
Includes a minor fixup to macro 2.0 parsing.

bors r+

### CODE REVIEW

File: crates/hir_def/src/nameres/collector.rs [hunk 1]
```diff
@@ -1402,8 +1402,18 @@ impl ModCollector<'_, '_> {
 
         // Case 1: builtin macros
         if attrs.by_key("rustc_builtin_macro").exists() {
+            // `#[rustc_builtin_macro = "builtin_name"]` overrides the `macro_rules!` name.
+            let name;
```
Comment (edwin0cheng) [inline crates/hir_def/src/nameres/collector.rs:1406]: Forgot to delete ?

**Comment by lnicola** [comment]
changelog fix

File: crates/hir_def/src/nameres/collector.rs (hunk 1 above)
Comment (bjorn3) [reply crates/hir_def/src/nameres/collector.rs:1406]: No, it is assigned a `tt::Ident` below at line 1410. The next `let name =` then gets a reference to the `tt::Ident` as value at line 1411.

Comment (edwin0cheng) [reply crates/hir_def/src/nameres/collector.rs:1406]: TIL ! Neat !

**Pull request merged**